independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/injection/order/InjectionOrderTest.java

```diff
@@ -0,0 +1,113 @@
+package io.quarkus.arc.test.injection.order;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import jakarta.enterprise.context.Dependent;
+import jakarta.inject.Inject;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.test.ArcTestContainer;
+
+public class InjectionOrderTest {
```

Comment:
This test actually does not verify that injection points from the superclass are injected before injection points from the subclass, or am I missing something?

Reply:
Humm, good point. Yeah, I don't think that is tested -- the test currently only verifies that fields are injected before initializer methods. I think I might have an idea how to test this (though the AtInject TCK is a lot more complete than this test will ever be :-) ).

Reply:
Unfortunately you can only capture constructor invocation and initializer method invocation, not field injection. The only possibility I'm aware of is checking whether fields are already injected or not in the constructor and especially in the initializer method.

Reply:
> The only possibility I'm aware of is checking whether fields are already injected or not in the constructor and especially in the initializer method.

Yep, makes sense.  If a field is injected then just add a missing "field injection event" in the constructor and initializer... It's not accurate but good enough for the testing purpose I think.

Reply:
Done.